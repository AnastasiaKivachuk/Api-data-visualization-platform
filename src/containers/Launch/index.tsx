import React from "react"

import Typography from "@mui/material/Typography";
import styled from "@emotion/styled";
import CheckIcon from '@mui/icons-material/Check';
import CancelIcon from '@mui/icons-material/Cancel';

import StyledCarousel from "../../components/StyledCarousel";
import {LaunchWithDetailsFullInterface} from "../../interfaces/lauch.interface";
import {getDateWithFormat} from "../../helpers/functions.helpers";
import UsefulLinks from "../../components/UsefulLinks";
import Grid from "@mui/material/Grid";

const ImgIcon = styled.img`
      margin: 0 auto;
    `

const StyledCheckIcon = styled(CheckIcon)`
      color: green;
    `

const StyledCancelIcon = styled(CancelIcon)`
      color: red;
    `

const Launch = ({launch}: LaunchWithDetailsFullInterface) => {
    return (
        <>
            <Grid container alignItems={'center'} justifyContent={'center'} gap='15px'>
                <Typography variant="h2" textAlign='center'>{launch.mission_name}</Typography>
                {launch.launch_success ?
                    <StyledCheckIcon/> : <StyledCancelIcon/>}
            </Grid>
            <Typography variant='h6' textAlign='center'>Date of
                launch: {getDateWithFormat(launch.launch_date_utc)}</Typography>
            <ImgIcon src={launch.links.mission_patch_small} alt={'logo'}/>
            <Typography variant='subtitle1' textAlign='center'>{launch.details}</Typography>
            <Typography variant='h3' textAlign='center'>Rocket: {launch.rocket.rocket_name}</Typography>

            <StyledCarousel imagesUrls={launch.links.flickr_images} alt={launch.rocket.rocket_name}/>
            <UsefulLinks launchLinks={launch.links}/>

        </>
    )
}

Launch.defaultProps = {
    launch: null
}

export default Launch
