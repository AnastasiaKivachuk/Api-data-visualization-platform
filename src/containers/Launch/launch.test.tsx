import {shallow} from "enzyme";
import React from "react";

import Launch from "./index";
import {LaunchWithDetailsFullInterface} from "../../interfaces/lauch.interface";

const MockLaunchData = (status = true) => ({
    details: "Second GTO launch for Falcon 9. The USAF evaluated launch data from this flight as part of a separate certification program for SpaceX to qualify to fly U.S. military payloads and found that the Thaicom 6 launch had \"unacceptable fuel reserves at engine cutoff of the stage 2 second burnoff",
    id: "13",
    launch_date_utc: "2014-01-06T18:06:00.000Z",
    launch_success: status,
    mission_name: "Thaicom 6",
    rocket: {
        rocket_name: "Falcon 9"
    },
    links: {
        article_link: "http://spacenews.com/38959spacex-delivers-thaicom-6-satellite-to-orbit/",
        flickr_images: ['https://farm9.staticflickr.com/8617/16789019815_f99a165dc5_o.jpg', 'https://farm8.staticflickr.com/7619/16763151866_35a0a4d8e1_o.jpg', 'https://farm9.staticflickr.com/8569/16169086873_4d8829832e_o.png'],
        mission_patch_small: "https://images2.imgbox.com/d8/6d/fnqIBEJh_o.png",
        video_link: "https://www.youtube.com/watch?v=AnSNRzMEmCU",
        wikipedia: "https://en.wikipedia.org/wiki/Thaicom_6"
    }
})

const setUp = (props: LaunchWithDetailsFullInterface) => shallow(<Launch {...props}/>);

describe("MockLaunchData component", () => {

    it("should render MockLaunchData component with status true", () => {
        const component = setUp({launch: MockLaunchData()})
        expect(component).toMatchSnapshot();
    })

    it("check all elements", () => {

        const component = setUp({launch: MockLaunchData()})
        const text = ['Thaicom 6', "Date of launch: 06.01.2014, 21:06 PM", "Second GTO launch for Falcon 9. The USAF evaluated launch data from this flight as part of a separate certification program for SpaceX to qualify to fly U.S. military payloads and found that the Thaicom 6 launch had \"unacceptable fuel reserves at engine cutoff of the stage 2 second burnoff", "Rocket: Falcon 9"]
        expect(component.find('ForwardRef(Typography)').length).toEqual(4)
        component.find('ForwardRef(Typography)').map((item, index) =>
            expect(item.text()).toEqual(text[index]))

    })

    it("should render MockLaunchData component with status false", () => {
        const component = setUp({launch: MockLaunchData(false)})
        expect(component).toMatchSnapshot();

    })
})
