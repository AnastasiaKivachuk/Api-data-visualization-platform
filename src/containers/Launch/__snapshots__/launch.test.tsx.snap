// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MockLaunchData component should render MockLaunchData component with status false 1`] = `
<Fragment>
  <ForwardRef(Grid)
    alignItems="center"
    container={true}
    gap="15px"
    justifyContent="center"
  >
    <ForwardRef(Typography)
      textAlign="center"
      variant="h2"
    >
      Thaicom 6
    </ForwardRef(Typography)>
    <Styled(Component) />
  </ForwardRef(Grid)>
  <ForwardRef(Typography)
    textAlign="center"
    variant="h6"
  >
    Date of launch: 
    06.01.2014, 21:06 PM
  </ForwardRef(Typography)>
  <Styled(img)
    alt="logo"
    src="https://images2.imgbox.com/d8/6d/fnqIBEJh_o.png"
  />
  <ForwardRef(Typography)
    textAlign="center"
    variant="subtitle1"
  >
    Second GTO launch for Falcon 9. The USAF evaluated launch data from this flight as part of a separate certification program for SpaceX to qualify to fly U.S. military payloads and found that the Thaicom 6 launch had "unacceptable fuel reserves at engine cutoff of the stage 2 second burnoff
  </ForwardRef(Typography)>
  <ForwardRef(Typography)
    textAlign="center"
    variant="h3"
  >
    Rocket: 
    Falcon 9
  </ForwardRef(Typography)>
  <StyledCarousel
    alt="Falcon 9"
    imagesUrls={
      Array [
        "https://farm9.staticflickr.com/8617/16789019815_f99a165dc5_o.jpg",
        "https://farm8.staticflickr.com/7619/16763151866_35a0a4d8e1_o.jpg",
        "https://farm9.staticflickr.com/8569/16169086873_4d8829832e_o.png",
      ]
    }
  />
  <UsefulLinks
    launchLinks={
      Object {
        "article_link": "http://spacenews.com/38959spacex-delivers-thaicom-6-satellite-to-orbit/",
        "flickr_images": Array [
          "https://farm9.staticflickr.com/8617/16789019815_f99a165dc5_o.jpg",
          "https://farm8.staticflickr.com/7619/16763151866_35a0a4d8e1_o.jpg",
          "https://farm9.staticflickr.com/8569/16169086873_4d8829832e_o.png",
        ],
        "mission_patch_small": "https://images2.imgbox.com/d8/6d/fnqIBEJh_o.png",
        "video_link": "https://www.youtube.com/watch?v=AnSNRzMEmCU",
        "wikipedia": "https://en.wikipedia.org/wiki/Thaicom_6",
      }
    }
  />
</Fragment>
`;

exports[`MockLaunchData component should render MockLaunchData component with status true 1`] = `
<Fragment>
  <ForwardRef(Grid)
    alignItems="center"
    container={true}
    gap="15px"
    justifyContent="center"
  >
    <ForwardRef(Typography)
      textAlign="center"
      variant="h2"
    >
      Thaicom 6
    </ForwardRef(Typography)>
    <Styled(Component) />
  </ForwardRef(Grid)>
  <ForwardRef(Typography)
    textAlign="center"
    variant="h6"
  >
    Date of launch: 
    06.01.2014, 21:06 PM
  </ForwardRef(Typography)>
  <Styled(img)
    alt="logo"
    src="https://images2.imgbox.com/d8/6d/fnqIBEJh_o.png"
  />
  <ForwardRef(Typography)
    textAlign="center"
    variant="subtitle1"
  >
    Second GTO launch for Falcon 9. The USAF evaluated launch data from this flight as part of a separate certification program for SpaceX to qualify to fly U.S. military payloads and found that the Thaicom 6 launch had "unacceptable fuel reserves at engine cutoff of the stage 2 second burnoff
  </ForwardRef(Typography)>
  <ForwardRef(Typography)
    textAlign="center"
    variant="h3"
  >
    Rocket: 
    Falcon 9
  </ForwardRef(Typography)>
  <StyledCarousel
    alt="Falcon 9"
    imagesUrls={
      Array [
        "https://farm9.staticflickr.com/8617/16789019815_f99a165dc5_o.jpg",
        "https://farm8.staticflickr.com/7619/16763151866_35a0a4d8e1_o.jpg",
        "https://farm9.staticflickr.com/8569/16169086873_4d8829832e_o.png",
      ]
    }
  />
  <UsefulLinks
    launchLinks={
      Object {
        "article_link": "http://spacenews.com/38959spacex-delivers-thaicom-6-satellite-to-orbit/",
        "flickr_images": Array [
          "https://farm9.staticflickr.com/8617/16789019815_f99a165dc5_o.jpg",
          "https://farm8.staticflickr.com/7619/16763151866_35a0a4d8e1_o.jpg",
          "https://farm9.staticflickr.com/8569/16169086873_4d8829832e_o.png",
        ],
        "mission_patch_small": "https://images2.imgbox.com/d8/6d/fnqIBEJh_o.png",
        "video_link": "https://www.youtube.com/watch?v=AnSNRzMEmCU",
        "wikipedia": "https://en.wikipedia.org/wiki/Thaicom_6",
      }
    }
  />
</Fragment>
`;
